internal/broker/config.go

```diff
@@ -24,15 +27,55 @@ const (
 
 	// usersSection is the section name in the config file for the users and broker specific configuration.
 	usersSection = "users"
+	// allowedUsersKey is the key in the config file for the users that are allowed to access the machine.
+	allowedUsersKey = "allowed_users"
+	// ownerKey is the key in the config file for the owner of the machine.
+	ownerKey = "owner"
 	// homeDirKey is the key in the config file for the home directory prefix.
 	homeDirKey = "home_base_dir"
 	// SSHSuffixKey is the key in the config file for the SSH allowed suffixes.
 	sshSuffixesKey = "ssh_allowed_suffixes"
+
+	// AllUsersKeyword is the keyword for the `allowed_users` key that allows access to all users.
+	AllUsersKeyword = "ALL"
+	// OwnerUserKeyword is the keyword for the `allowed_users` key that allows access to the owner.
+	OwnerUserKeyword = "OWNER"
+
+	// ownerAutoRegistrationConfigPath is the name of the file that will be auto-generated to register the owner.
+	ownerAutoRegistrationConfigPath     = "20-owner-autoregistration.conf"
+	ownerAutoRegistrationConfigTemplate = "templates/20-owner-autoregistration.conf.tmpl"
+)
+
+var (
+	//go:embed templates/20-owner-autoregistration.conf.tmpl
+	ownerRegistrationConfig embed.FS
 )
 
+type templateEnv struct {
+	Owner string
+}
+
+type userConfig struct {
+	clientID     string
+	clientSecret string
+	issuerURL    string
+
+	allowedUsers          map[string]bool
+	allUsersAllowed       bool
+	ownerAllowed          bool
+	firstUserBecomesOwner bool
+	owner                 string
+	homeBaseDir           string
+	allowedSSHSuffixes    []string
+}
+
+func getDropInDir(cfgPath string) string {
+	return cfgPath + ".d"
```

Comment:
We may have this already, so let's share this logic in the same place for the 3 (now) occurrencies

Reply:
I will leave the function in this file and make exportable, unless you have any objections with that.